### ISSUE TITLE
Dendrogram.scale_bar automated precision needs improvement

### ISSUE THREAD

**Issue by GavinHuttley** [cogent3/cogent3#560]
for the following tree `'(Sbay:0.012690237171196569,(Skud:0.011606490743365165,(Smik:0.008462391575411454,(Spar:0.004457756840636469,Scer:0.005227562388409443):0.0028390886420649655):0.0037881351326451394):0.00043971046313997543);'`, shows scale bar with length 0.

![bug](https://user-images.githubusercontent.com/3102996/76039515-c869bb80-5fa0-11ea-9de1-99ea1f45805b.png)

**Comment by u6052029** [comment]
Is there a code example where I can run this tree in Jupyter lab?

**Comment by GavinHuttley** [comment]
```python
from cogent3 import make_tree

tree = make_tree('(Sbay:0.012690237171196569,(Skud:0.011606490743365165,(Smik:0.008462391575411454,(Spar:0.004457756840636469,Scer:0.005227562388409443):0.0028390886420649655):0.0037881351326451394):0.00043971046313997543);')
tree.get_figure(style="angular")
```

**Comment by u6052029** [comment]
It displays 0.00 because the value it is trying to display is "0.0012690237171196569"

Should we display to 3 decimal places or some other strategy?

**Comment by GavinHuttley** [comment]
We need to adapt the display to the numbers magnitude. For instance, in this case, we might do "{:,.1e}".format(n).

Set a cut-off for switching to scientific notation when number < 0.01...

----------------------------------
Professor Gavin Huttley,
Research School of Biology,
ANU College of Science,
RN Robertson Building,
46 Sullivans Creek Road,
The Australian National University
Canberra ACT 2600 Australia

+61 2 6125 5308

http://biology.anu.edu.au/people/gavin-huttley

CRICOS #00120C
----------------------------------


On Thu, Jul 23, 2020 at 1:49 PM, Moses Koh <notifications@github.com> wrote:

Is there a code example where I can run this tree in Jupyter lab?

**Comment by GavinHuttley** [comment]
fixed in 302c496617b383b90a96cb70517dcf2112ff5bac
